### PR TITLE
enable server-side validation by default

### PR DESCRIPTION
**Pull request by ayj**

### CODE REVIEW

**Comment by mandarjog** [comment]
@ayj please see https://github.com/istio/istio/issues/6031 
Under certain conditions mixer validator is interfering.

**Comment by ayj** [comment]
Thanks @mandarjog. I removed k8s APA from the list of validated types for now while we sort out the mixer validation issue. In the meantime, it's still worthwhile to validate all other working types.

**Comment by ayj** [comment]
Chatted with @guptasu offline. He has a PR that cleans up a lot of the mixer validation code. It should land sometime this week. We'll revisit the disabled mixer type after that PR is merged.

**Comment by ayj** [comment]
> To confirm: galley is self-configuring, like the sidecar injector ? Does it use the same mechanism ?

Yes, for better or worse.

**Comment by morvencao** [comment]
@ayj Simply enable `galley`& disable validating `kubernetes` CR, the mixer hook job still keeps failing:
```
root@master:~# kubectl -n istio-system logs -f istio-mixer-post-install-s9whg
2018-06-06 03:15:37.405274 I | proto: duplicate proto type registered: google.protobuf.Any
2018-06-06 03:15:37.405457 I | proto: duplicate proto type registered: google.protobuf.Duration
2018-06-06 03:15:37.405611 I | proto: duplicate proto type registered: google.protobuf.Timestamp
stdio "handler" configured
prometheus "handler" configured
kubernetesenv "handler" configured
rule "kubeattrgenrulerule" configured
kubernetes "attributes" configured
destinationrule "istio-policy" configured
destinationrule "istio-telemetry" configured
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: 1 error occurred:

* failure on attributes.kubernetes.istio-system with the new manifest: error evaluating AttributeBinding expression 'adapter_template_kubernetes.output.destination_pod_ip | ip("0.0.0.0")' for attribute 'destination.ip': unknown attribute adapter_template_kubernetes.output.destination_pod_ip
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: 1 error occurred:

* failure on attributes.kubernetes.istio-system with the new manifest: error evaluating AttributeBinding expression 'adapter_template_kubernetes.output.destination_service_account_name | "unknown"' for attribute 'destination.serviceAccount': unknown attribute adapter_template_kubernetes.output.destination_service_account_name
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Variables[userAgent]'; unknown attribute request.useragent
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: 1 error occurred:

* actions[0].instances[0]: accesslog.logentry not found
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Dimensions[source_version]'; unknown attribute source.labels
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Value'; unknown attribute response.duration
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Value'; unknown attribute request.size
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Value'; unknown attribute response.size
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Value'; unknown attribute connection.sent.bytes
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: failed to evaluate expression for field 'Value'; unknown attribute connection.received.bytes
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: 4 errors occurred:

* actions[0].instances[0]: requestcount.metric not found
* actions[0].instances[1]: requestduration.metric not found
* actions[0].instances[2]: requestsize.metric not found
* actions[0].instances[3]: responsesize.metric not found
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: 2 errors occurred:

* actions[0].instances[0]: tcpbytesent.metric not found
* actions[0].instances[1]: tcpbytereceived.metric not found
Error from server: error when creating "/tmp/mixer/custom-resources.yaml": admission webhook "mixer.validation.istio.io" denied the request: 1 error occurred:

* match: unknown attribute context.protocol
```

/cc @gyliu513

**Comment by ayj** [comment]
@morvencao, mixer validation for `kuberneteses` resources seems to have a bug. I've disabled validation for that now  
[here](https://github.com/istio/istio/pull/6035/commits/2ba9e3a2d04d4bdbbfc62af9d8bf2121e9574be0#diff-d275ef1428761dab6c1b937877bf7fe0L86). You should be able to `kubectl edit validatingwebhookconfiguration istio-galley` and remove the `kuberneteses` from the list of mixer resources. 

See https://github.com/istio/istio/issues/6055 for tracking issue.

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -238,7 +238,7 @@ sidecarInjectorWebhook:
 # galley configuration
 #
 galley:
-  enabled: false
+  enabled: true
```
Comment (costinm) [inline install/kubernetes/helm/istio/values.yaml]: Please add _lots_ of comments. Users don't know or care about internal component names.

**Comment by costinm** [comment]
Even MTLS is opt-in / disabled by default.  I haven't looked at the code details - if this is just about running galley and some per-namespace label is used, no problem, as I mentioned. But unless I miss-understood how it works - it is far too risky and far too little tested.

**Comment by mandarjog** [comment]
The only thing galley enables is config validators, is that correct? Are there any other parts that get turned on and affect behaviour ?

**Pull request merged**